File: backend/app/Models/Player.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Player extends Model
{
    use HasFactory;

    protected $fillable = ['name', 'number', 'position', 'role', 'team_id', 'category_id'];

    public function team()
    {
        return $this->belongsTo(Team::class);
    }

    public function positions()
    {
        return $this->hasMany(PlayerPosition::class);
    }

    public function tacticalInstructions()
    {
        return $this->belongsToMany(TacticalInstruction::class, 'instruction_player');
    }

    public function category()
    {
        return $this->belongsTo(Category::class);
    }
}

File: frontend/lib/models/player.dart
class Player {
  final int id;
  final String name;
  final int number;
  final String position;
  final String role;
  final int teamId;
  final int? categoryId;
  final String? categoryName;
  final DateTime? createdAt;

  Player({
    required this.id,
    required this.name,
    required this.number,
    required this.position,
    required this.role,
    required this.teamId,
    this.categoryId,
    this.categoryName,
    this.createdAt,
  });

  factory Player.fromMap(Map<String, dynamic> map) {
    int toInt(dynamic v) {
      if (v is int) return v;
      if (v is String) return int.tryParse(v) ?? 0;
      return 0;
    }

    return Player(
      id: map['id'] as int,
      name: map['name'] as String? ?? '',
      number: toInt(map['number']),
      position: map['position'] as String? ?? '',
      role: map['role'] as String? ?? '',
      teamId: toInt(map['team_id']),
      categoryId: map['category_id'] != null ? toInt(map['category_id']) : null,
      categoryName: map['category'] is Map<String, dynamic>
          ? (map['category']['name'] as String?)
          : map['category_name'] as String?,
      createdAt: map['created_at'] is String ? DateTime.tryParse(map['created_at'] as String) : null,
    );
  }
}
